code/ship/shield.cpp

```diff
@@ -487,36 +490,43 @@ void render_shield(int shield_num)
 	// don't try to draw if we don't have an ani
 	if ( sa->first_frame >= 0 )
 	{
-		frame_num = bm_get_anim_frame(sa->first_frame, f2fl(Missiontime - Shield_hits[shield_num].start_time), f2fl(SHIELD_HIT_DURATION));
+		frame_num = bm_get_anim_frame(sa->first_frame, f2fl(Missiontime - sh->start_time), f2fl(SHIELD_HIT_DURATION));
 		bitmap_id = sa->first_frame + frame_num;
 
 		float alpha = 0.9999f;
 		nebula_handle_alpha(alpha, centerp, Neb2_fog_visibility_shield);
 
 		ubyte r, g, b;
-		r = (ubyte)(Shield_hits[shield_num].rgb[0] * alpha);
-		g = (ubyte)(Shield_hits[shield_num].rgb[1] * alpha);
-		b = (ubyte)(Shield_hits[shield_num].rgb[2] * alpha);
+		r = (ubyte)(sh->rgb[0] * alpha);
+		g = (ubyte)(sh->rgb[1] * alpha);
+		b = (ubyte)(sh->rgb[2] * alpha);
 
 		if ( bitmap_id <= -1 ) {
 			return;
 		}
 
 		if ( (Detail.shield_effects == 1) || (Detail.shield_effects == 2) ) {
-			shield_render_low_detail_bitmap(bitmap_id, alpha, &Global_tris[Shield_hits[shield_num].tri_list[0]], orient, centerp, r, g, b);
+			shield_render_low_detail_bitmap(bitmap_id, alpha, &Global_tris[sh->tri_list[0]], orient, centerp, r, g, b);
 		} else if ( Detail.shield_effects < 4 ) {
-			for ( int i = 0; i < Shield_hits[shield_num].num_tris; i++ ) {
-				shield_render_triangle(bitmap_id, alpha, &Global_tris[Shield_hits[shield_num].tri_list[i]], orient, centerp, r, g, b);
+			for ( int i = 0; i < sh->num_tris; i++ ) {
+				shield_render_triangle(bitmap_id, alpha, &Global_tris[sh->tri_list[i]], orient, centerp, r, g, b);
 			}
 		} else {
 			float hit_radius = pm->core_radius;
 			if ( si->is_big_or_huge() ) {
 				hit_radius = pm->core_radius * 0.5f;
 			}
 
+			if (sh->radius_override >= 0.0f)
+				hit_radius = sh->radius_override;
+
+			if (si->max_shield_impact_effect_radius >= 0.0f && hit_radius > si->max_shield_impact_effect_radius) {
+				hit_radius = si->max_shield_impact_effect_radius;
+			}
```

Comment:
Is 0 a valid value here?  A 0-radius decal would be effectively no decal.  And maybe that's what the modeler wants, but sometimes draw functions throw errors if the coder tries to render a 0-width or 0-radius thing.

Reply:
Yeah fair enough, I'll just skip the decal render if radius is 0, and keep everything else the same to ensure modders can use it to disable the effect or something.